src/Sema.zig

```diff
@@ -3275,16 +3288,7 @@ fn ensureResultUsed(
             };
             return sema.failWithOwnedErrorMsg(msg);
         },
-        else => {
-            const msg = msg: {
-                const msg = try sema.errMsg(block, src, "value of type '{}' ignored", .{operand_ty.fmt(sema.mod)});
-                errdefer msg.destroy(sema.gpa);
-                try sema.errNote(block, src, msg, "all non-void values must be used", .{});
-                try sema.errNote(block, src, msg, "this error can be suppressed by assigning the value to '_'", .{});
-                break :msg msg;
-            };
-            return sema.failWithOwnedErrorMsg(msg);
-        },
+        else => return sema.failValueIgnored(block, src, operand_ty),
```

Comment:
The `error is ignored` part should also be present for comptime functions calls for consistency. A better solution would be to make `ensureResultUsed` accept a `Type` instead of an `Air.Inst.Ref`.